perceval/backends/core/hyperkitty.py

```diff
@@ -221,13 +221,10 @@ def _parse_date_from_filepath(self, filepath):
 
         return dt
 
-    def _download_archive(self, url, params, filepath):
-        r = requests.get(url, params=params, stream=True)
-        r.raise_for_status()
-
+    def _read_archive(self, raw_content, url, filepath):
```

Comment:
Why don't you keep it as before? Just `_download archive` (downloading the content inside the method). I don't see the reason for changing it.